packages/core/src/typeschema/validation.ts

```diff
@@ -510,6 +514,41 @@ class ResourceValidator implements CrawlerVisitor {
     }
   }
 
+  /**
+   * Validate FHIR base64Binary primitive.
+   * - No generic string checks (whitespace trimming or 1MB limit) apply.
+   * - Validate base64 alphabet and padding.
+   * - Apply an approximate decoded-size limit to guard against pathological payloads.
+   * @param str - The base64-encoded string.
+   * @param path - The FHIR element path for issue reporting.
+   */
+  private validateBase64Binary(str: string, path: string): void {
+    // Must be a non-empty string; allow empty only if schema constraints permit it (handled elsewhere)
+    if (typeof str !== 'string') {
+      this.issues.push(createStructureIssue(path, 'Invalid JSON type: expected string, but got ' + typeof str));
+      return;
+    }
+
+    const regex = validationRegexes.base64Binary;
+    if (regex && !regex.exec(str)) {
```

Comment:
Is it possible for the regex to be falsy here?
```suggestion
    if (!validationRegexes.base64Binary.exec(str)) {
```